### ISSUE TITLE
glottolog import as glottocode_by_iso is an empty dictionary

### ISSUE THREAD

**Issue by LinguList** [glottobank/lexibank-data#75]
[provided I understand the code properly]

In this line, dataset.glottocode_by_iso is declared as empty dictionary

* https://github.com/glottobank/lexibank-data/blob/master/pylexibank/dataset.py#L85

this yields empty glottocodes in all cases like tukanoaligned, were we only have iso and want glottocode.

I do not sufficiently understand pyglottolog api, which is why I file this as an ordinary issue, since I did not find any function or smart way to get iso-codes-to-glotto-code-conversion from the code there (using the api.Glottolog takes ages to load...).

sorry for being not more helpful...

**Comment by xrotwang** [comment]
Yeah, it's a bit of a hack - I'm sorry. `glottocode_by_iso` does only contain data when the `cldf` command is run, see [here](https://github.com/glottobank/lexibank-data/blob/master/pylexibank/cli.py#L148-L151). I needed the glottolog data in the `cldf` functions, but since it's expensive to compute didn't want to recreate it per dataset (and didnt't want to compute it when not necessary). So it ended up where it is now. We should talk about a better design next weekend.

**Comment by xrotwang** [comment]
AFAIKT it does work now for `tukanoaligned`, so can we close this?

**Comment by LinguList** [comment]
it doesn't work on my machine, but I don't trust my py3 version of glottolog. But do you also make the iso2glotto in the line? I only see you assigning glotto.id as key? Yet if tukano works and glottocodes are created right now, let's close this and I'll check my  specific problems during the weekend.

**Comment by LinguList** [comment]
sorry, I see, this should really do the trick, didn't really catch up with the code. so it's my computer problem here, maybe some py3 stuff...

**Comment by xrotwang** [comment]
As far as I remeber, in your installation there were problems for `pylottolog` locating the repository, right?

**Comment by LinguList** [comment]
well, that may be the problem, but when I point explicitly to glottolog
now, it takes ages to load, so long I did even stop trying. I guess it
was some ini-file or yaml that took so long (I remember glottolog has
one), but is more than 2 minutes for that normal (I stopped after two
minutes because I lost patience). And when I use the normal
glottolog.api, it loads languoids, still needs ages to create the
dictionary, if I try explicitly, but it knows the repo (I re-installed now).
